nityam_kumar/PostgresAssignment/src/controller/mark.ts

```diff
@@ -0,0 +1,165 @@
+import { Context } from "vm";
+import * as marksService from "../services/mark";
+
+import markSchema from "../db/validateSchema/markSchema";
+import AppError from "../utils/appError";
+
+interface IMark {
+  student_id: string;
+  subject_id: string;
+  marks: number;
+  teacher_id: string;
+  class_number: number;
+}
+
+export const createMarks = async (ctx: Context) => {
+  const m1: IMark = ctx.request.body;
+  try {
+    await markSchema.validateAsync(m1);
+
+    await marksService.checkStudentExist(m1);
+
+    await marksService.checkSubjectExist(m1);
+
+    await marksService.checkAlreadyExist(m1);
+
+    await marksService.addMarkDB(m1);
+    ctx.status = 200;
+    ctx.body = {
+      status: `successfully created marks with ${m1.student_id} & ${m1.subject_id}`,
+    };
+  } catch (err) {
+    if (!err.status) throw new AppError(err.message, 400);
```

Comment:
Please use different try/catch for each case.

Reply:
added different try catch for each case